blueocean-dashboard/src/main/js/creation/bitbucket/cloud/steps/BbCloudCompleteStep.jsx

```diff
@@ -0,0 +1,122 @@
+        } else if (state === STATE.STEP_COMPLETE_EVENT_TIMEOUT) {
+            return t('creation.core.status.pending');
+        } else if (state === STATE.STEP_COMPLETE_MISSING_JENKINSFILE) {
+            return <span>{t('creation.core.error.missing.jenkinsfile')} <i>{repo.name}</i></span>;
+        } else if (state === STATE.STEP_COMPLETE_SUCCESS) {
+            return t('creation.core.status.completed');
+        }
+
+        return t('creation.core.error.unexpected');
+    }
+
+    _getError(state) {
+        return state === STATE.STEP_COMPLETE_SAVING_ERROR ||
+            state === STATE.STEP_COMPLETE_EVENT_ERROR;
+    }
+
+    _getContent(state) {
+        const { redirectTimeout, pipelineName } = this.props.flowManager;
+
+        let copy = '';
+        let showDashboardLink = false;
+        let showCreateLink = false;
+
+        if (state === STATE.STEP_COMPLETE_SAVING_ERROR) {
+            copy = t('creation.core.error.saving.pipeline');
+        } else if (state === STATE.STEP_COMPLETE_EVENT_ERROR) {
+            copy = t('creation.core.error.creating.pipeline');
+            showDashboardLink = true;
+        } else if (state === STATE.STEP_COMPLETE_EVENT_TIMEOUT) {
+            copy = t('creation.core.status.waiting');
+            showDashboardLink = true;
+        } else if (state === STATE.STEP_COMPLETE_MISSING_JENKINSFILE) {
+            showCreateLink = true;
+        } else if (state === STATE.STEP_COMPLETE_SUCCESS) {
+            setTimeout(() => this.navigatePipeline(), redirectTimeout);
+        }
+
+        return (
+            <div>
+                <p className="instructions">{copy}</p>
+
+                { showDashboardLink &&
+                <div>
+                    <p>{t('creation.core.status.return.new_pipelines')}.</p>
+
+                    <button onClick={() => this.navigateDashboard()}>{t('creation.core.button.dashboard')}</button>
+                </div>
+                }
+
+                { showCreateLink &&
+                <div>
+                    <Extensions.Renderer extensionPoint="jenkins.pipeline.create.missing.jenkinsfile"
+                                         organization={'jenkins'} fullName={pipelineName}
+                    />
+                </div>
+                }
+            </div>
+        );
+    }
+
+    render() {
+        const { flowManager } = this.props;
+        const status = this._getStatus(flowManager.stateId, this.props.status);
+        const error = this._getError(flowManager.stateId);
+        const title = this._getTitle(flowManager.stateId, flowManager.selectedRepository);
+        const content = this._getContent(flowManager.stateId);
+
+        return (
+            <FlowStep {...this.props} className="github-complete-step" title={title} status={status} error={error}>
+                {content}
+            </FlowStep>
+        );
+    }
+}
+
+BbCloudCompleteStep.propTypes = {
+    flowManager: PropTypes.object,
+    status: PropTypes.string,
+};
```

Comment:
Should this step actually go in `/bitbucket/steps` ? I believe it's shared by both cloud and server?